org.eclipse.winery.repository.ui/src/app/instance/serviceTemplates/threatAssessment/threatAssessment.component.ts

```diff
@@ -0,0 +1,122 @@
+/*******************************************************************************
+ * Copyright (c) 2018 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0, or the Apache Software License 2.0
+ * which is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+ *******************************************************************************/
+import { Component, OnInit, Input } from '@angular/core';
+import { ThreatAssessmentService } from './threatAssessment.service';
+
+export interface QName {
+    localPart: string;
+    namespace: string;
+}
+
+export interface ThreatProperties {
+    description?: string;
+}
+
+export interface ThreatTarget {
+    nodeType: string;
+    nodeTemplate: string;
+}
+
+export interface ThreatInterface {
+    name: string;
+    mitigations?: [QName];
+    severity?: string;
+    properties?: ThreatProperties;
+    targets?: [ThreatTarget];
+}
+
+@Component({
+    selector: 'winery-threat-model',
+    templateUrl: './threatAssessment.component.html',
+    providers: [ThreatAssessmentService]
+})
+
+export class ThreatAssessmentComponent implements OnInit {
+    @Input() threatAssessmentData = true;
+    loading = true;
+    threats: ThreatInterface[];
+    svnfs: QName[];
+
+    s2Q = ThreatAssessmentComponent.stringToQName;
+
+    static stringToQName(name: string): QName {
```

Comment:
Since this is a util method for a QName, this could be part of the QName class.

Reply:
added to newly created QName class